async.go

```diff
@@ -3,25 +3,140 @@ package async
 import (
 	"context"
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/netbookai/log"
 	"github.com/netbookai/log/loggers"
 	"github.com/pkg/errors"
 )
 
-func run(ctx context.Context, timeout time.Duration, fn func(context.Context)) error {
-	errCh := make(chan error)
+type taskGroup struct {
+	tasks []task
+}
+
+type task struct {
+	name        string
+	taskWrapper func(context.Context) error
+}
+
+func New() *taskGroup {
+	return &taskGroup{}
+}
+
+func (g *taskGroup) Add(name string, taskWrapper func(context.Context) error) *taskGroup {
```

Comment:
why return, we have a pointer receiver, any reason to return ?

Reply:
this would help if we are chaining, but would be too long statement if we decide to create 2 or more tasks, 
good to avoid this IMO. what you think ?

Reply:
yeah for chaining only...rather than having more than one statements ...i think chaining is good. See examples